src/data/DataSetExecutionD2Repository.ts

```diff
@@ -1,21 +1,21 @@
-import { D2Api } from "@eyeseetea/d2-api/2.36";
+import { D2Api } from "types/d2-api";
 import { Async } from "domain/entities/Async";
 import { DataSetExecutionRepository } from "domain/repositories/DataSetExecutionRepository";
-import { NotificationStore } from "domain/usecases/SendNotificationDataValuesUseCase";
-import { SettingsOptions } from "domain/repositories/SettingsRepository";
-
-const key = "executions";
+import { DataSetExecution } from "domain/usecases/SendNotificationDataValuesUseCase";
+import { extractNameSpaceAndKeyFromPath, SettingsOptions } from "domain/entities/Settings";
```

Comment:
The entity knows nothing about dataStore namespace/key, that's a dhis2 detail, so the function should be implemented in this data repo.